### DISCUSSION TITLE
[WANTED] Pyhton EZSP Sniffer

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#692] [Q&A] [answered]
I  is using "Billys" (IKEA Zigbee modules with EZSP NCP firmare) as coordinators and sniffers  with [com.zsmartsystems.zigbee.sniffer](https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer).
Its working very well sniffing on my Windows 10 PRO and have doing over 100 hours sniffing in 5 different Zihgbee networks (also my IKEA GW after the problem getting the network key was solved).

After getting tuya ZBGW up and running and perhaps getting  the second comport working (its do and can mapping it to  one network port but the kernel is reclaiming it for local login then its detecting its receiving traffic on it) is i missing the function of using network mapped ports for sniffing like bellows is doing then running one NCP as coordinator.

I wold like having one python module for doing the sniffing and piping packages / network frames to wireshark so its possible using network ports from one tuya ZBGW or one Sonoff ZBB .
If its possible making as one docker image its being better but then its loosing the possible using local comports under Windows hosts but it worth it.

Perhaps its possible doing the same implanting with other coordinators and making it as one very open implementation using TI CC and all NCP that have support for sniffing (with standard or dictated sniffing firmware in the NCP).

I have before doing one request for updating the [Hue Thief](https://github.com/vanviegen/hue-thief) that is using one old bellows but looks being broke in newer python version and i cant getting it working and have not finding any old docker container that is not broke on the net.

Also doing one update of [husbzb-firmware](https://github.com/walthowd/husbzb-firmware) update utility to supporting network ports and also EFR32 chips wold being one great thing so users can doing firmware upgrading in there system by only installing one docker image.

Making separate docker containers that can being started after HA container is stopped and perhaps reading the comports config from HAs configuration and / or one local configuration file for doing it easy for not so experience user.

More ideas for good utility functions to implanting it in ZHA or as separate standalone containers is very welcome.

Is it some good python coders around that can helping getting this being implanted so we can using this great futures in our systems ?

Im not one cod warrior so i cant doing it (perhaps cut and pasting some device ID to one quirk is my level if black is not putting my down) :-((

More ideas and feedback is more then  welcome !!

**Comment by Adminiuga** [comment, marked as answer]
`belows dump --help`

**Reply by MattWestb** [reply]
Ops i was reading that for some weeks ago but i think i was forgetting it !!
Must trying it on one tuya ZBGW is i can getting it working.

Thanks for info !!

**Reply by Hedda** [reply]
Can bellows also provide a Zigbee sniffer interface for Wireshark similar to the [com.zsmartsystems.zigbee.sniffer](https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer) driver?

https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer/blob/master/README.md

https://wiki.wireshark.org/IEEE_802.15.4

https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

https://github.com/homewsn/whsniff#how-to-use-locally

**Reply by MattWestb** [reply]
I dont knowing if it can doing one god piping to wireshark but i was reading the dump is implanted and i like testing it and if its sable as Zsm sniffer (I is doing 25 hours sniffs for the moment of IKEA controllers for getting the battery draining problem under control).

tuya ZBGW have 2 comports and both is possible mapping to network ports but the second one (the "free" on) is being hocked of the kernel then being used and trying getting the login prompt for the local console so its not usebal for the moment but hopeful its getting solved in the future.

**Reply by MattWestb** [reply]
I was changing the port number in `.storage/core.config_entries` to my test  tuya ZBGW and restarted HA.
HA is starting but not bellows that is making i can using the CLI in the HA container to running below in CLI mode.
Then issue `bellows -d socket://192.168.2.60:8888 dump -c 25 -w /config/dmp.cap` bellows is making one cap file and i can opening it with wireshark and have all frames very nicely only problem its no pipe file so i must clicking reloading for reading the new frames that have coming in.

So its looks working great on the bellows side.
Only 2 problems was hitting my to day.
I was buying one new wireless keyboard and testing it in the same USB hub and the USB-TTL adapter using for sniffing was fired :-((
And also my LIDL tuya ZBGW (I have one older with labelled that i running in the test) was sadden stop sending beacons and i was trying connecting to it with SSH that have working but its refusing to connect. Using one other USB-TTL and i getting the command prompt for login but the root password that was extracted and working is not working.
So 2 of my possibility of sniffing have going up in smoke for one new wireless keyboard THAT IS UNFAIR !!

The good thing i still have some working USB-TTL adapters and one tuya ZBGW that is running well on my test system and i can sniffing it OK (but not with bellows for the moment then need its for the test instance).

Thanks @Adminiuga for the hit and if some have experience of setting up "real" pipes from the file to wireshark its very welcome to trying it out.

**Reply by tube0013** [reply]
Have you looked at using PTI in the SiLabs Firmware? On my the Series 2 boards I compiled the FW with PTI available on a 2 pin header on the board. https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2018/12/26/why_should_i_enable-jdZk

I haven't tried it yet, but it looks very appealing for analyzing traffic.

**Reply by MattWestb** [reply]
I have seen that the Zigbee module in the tuya ZBGW have it on the data chest but i think not in the cooked firmware.

I think i shall asking Gary how to using it because the "normal"  EZSP sniffing you only getting frames that is OK and all is not nice frames is ending up in dev null.

I need to see if its corrupted frames from some side that is making the router (coordinator mostly) is not acking pull requests from sleeping end devices.

I have some new strange sniffs but i posting them in the discussion its belongs.

By the was very vice "weather crazy" dog you is having that i like !!

**Reply by MattWestb** [reply]
@tube0013 I was asking Gary (our EZSP firmware cocking guru) and hi was only knowing that the PIT is working on Silabs dev kits that have the pins for reviving the signals from the module you are debugging and using Simplicity studios network analyzer.
Perhaps its possible with "other" hardware but i think the best is baying one of the cheapest dev kit if need doing deep package analyzes.

The marathon sniffing of IKEA remotes is not bringing any more then its possible doing some workaround to getting them working well and not draining there batteries so no need putting the money on it for that.

But it was interesting to see how the Zigbee  mesh is working at the lower level.

Perhaps one dev kit can being good as one birthday present to my self and trying doing one Zigbee controller bridge or updated IKEA remotes with more switch endpoints for being putted behind normal light power switches.

**Reply by MattWestb** [reply]
@tube0013 and other devs, wot dev king have you buying the SLWSTK6021A with 2 not so god radios or the SLWSTK6104A with one new nice  (also not so god radio) EFR32MG22 module ?

I think its better having 2 EFR32MG22 6 dBm modules to playing with than one 8 dBm with the MGM22P and all have 512 kB flash and 32 kB RAM = Zigbee R23 compatible.

And both kits is getting access to the Zigbee stack ??

Thanks for advises !  !

**Reply by tube0013** [reply]
I bought the SLWSTK2021A with the 2 test boards (I don't know if the other was available then) to get access to the SDK, have hardly used it beyond that. I had previous a Segger J-Link-mini edu which is less cumbersome to use than the dev board so I use that mainly.

**Reply by MattWestb** [reply]
Thanks for advice !

I have birthday in 6 mouths but i think its god as "covid 21 therapy" before the summer and (hopefully) holiday in southern Spain :-))

I have also bricked 2 IKEA modules that i was not possible getting back after SC was not writing the boot loader (and was writing app in the hardware aria) in the and is likely one hard ware brick but perhaps the Silabs J-Tag can fixing them.

I think i also going for the SLWSTK2021A so must looking how the delivery is in EU and worming the credit card up.

**Reply by Adminiuga** [reply]
did you mean SLWSTK6021A ? It gives you access to EmberZNet, but the boards that come with it are MG22 based and don't have NCP example project for it. So you would need to purchase one based on MG21 chip.

**Reply by MattWestb** [reply]
Thanks for info !

Sorry i was miss printing, SLWSTK6021A is with 2 boards and looks good to my.
(hawk eye Adminiuga !!)

I think i not going to using them as coordinators and if i like i can asking Gary for doing one NCP firmware for it then the SDK is supporting it.

I like trying doing one "multi switch" with 2 or 3 dimmer functions and perhaps using one "old IKEA module" or one E1743 that have one switch pair but its possible using the RX and TX test pads for the second one.

And i have one LIDL LED stripe have one nice MG21 that is easy getting hocked up and i think its being nice as NCP as only doing some bad RGBWW things :-))

And see wot it is bringing :-))

**Reply by Hedda** [reply]
Those interested in that official $99 SLWSTK6021A started kit probably also want to buy article SLWRB4180A for around $49 at the same time as that is the matching EFR32MG21 based high-power (+20 dBm) radio board for that dev board kit.

https://www.silabs.com/products/development-tools/wireless/slwrb4180a-efr32-wireless-gecko-radio-board

Google search shows that SLWRB4180A can be bought from many of resellers globally who also stock the SLWSTK6021A kit.

Update: "This board is obsolete and it has been replaced with SLWRB4180B. The replacement board has a different pin assignment which routes the push buttons to pins which support EM2 wake-up.":

https://www.silabs.com/development-tools/wireless/slwrb4180b-efr32xg21-wireless-gecko-radio-board

**Reply by MattWestb** [reply]
Its on the way :-))
```
Wednesday, March 24, 2021
--
5:40 PM | MANSFIELD, TX | Left FedEx origin facility
2:30 PM | GRAND PRAIRIE, TX | Picked up
```
But FX is horrible in Austria so hope its going OK all the way.

**Reply by MattWestb** [reply]
@tube0013 or some one that have on SM-011 module and and Simplicity Studio can reading the CTune value in the chip and /or in the manufacturing token ?
One KB how is being done in SS4: https://www.silabs.com/community/wireless/bluetooth/knowledge-base.entry.html/2019/01/22/saving_ctune_valuea-qCPl

I suspecting that the SM-011 module and the new Sonoff USB Zigbee 3 stick is **not** calibrated and have problems with staying on the correct frequency and reading CTune can saying if they have storing somthing or not in the chip. And the manufacture is trying patching the config in the EZSP firmware for compensating the bad done work.

The procedure for doing on real calibrating if some is interested:
https://www.silabs.com/community/wireless/proprietary/knowledge-base.entry.html/2019/03/18/hfxo_capacitor_bank-7uRt
 
If my thinking its true it can resulting in that devices (sleeping end device mostly) is heaving problem communicating with the coordinator then its frequency is miss tuned or need re tuning its radio all the time for keeping the radio channel open.

If its possible with one factors firmware that is signed and the chip is not erased and not with EZSP 6.7.9.0 that have some "fixed configuration" but all reading / info is more then welcome !!

The Silabs modules IKEA and Tube is using shall being OK from the factory if i have understanding all OK but i dont now its with tuya having doing it OK or not on there EF32MG2X modules.


Edit: One more interesting thing it can being bad using high power setting then it can doing the the RF drifting of the heating in the AP and getting problems with come devices that is tuned in the other direction ( on + x and the other - y) and getting problems.

**Reply by MattWestb** [reply]
Latest new is that the firmware maker is:
> If you compile the firmware yourself, please set the HFXO CTUNE value to 128.

So my theory is confirmed !!!!
The SM-011 and the USB stick is not calibrated or they have changing components that is miss tuning the RF radio !!!

**Reply by Hedda** [reply]
@MattWestb Maybe you could report this to ITead via https://support.itead.cc/support/home ?

Otherwise, I read that the project managers for it are zoe.zheng@itead.cc and jerry.shi@itead.cc

**Reply by MattWestb** [reply]
The Sonoff / itead devs / manufacturing is having large problems that is doing all being one mess for the moment.
Many Sonoff ZBB is having hardware problems with the hardware like broken contact between the EZSP module and the ESP and the PCB line is being broken between the layers of the PCB. The shouldering of the RF components is don with no good prescription and if getting little more or less led on the pads is doing the RF part being miss tuned and in combination with not calibrating the chip/module (with CTune) in the manufacturing process and trying compensating it with patching the firmware is giving my the feeling of how no name company is doing 5€ CC-2531 devices.

In combination with  the EFR322x is one large evolution and the the MG22 is having one separate processor for the radio part (that is good) but is still having problems getting all things working well all the way for getting one stable system (Silabs have not relesed one working NCP for the MG22 yet and i waiting for it being implanted for testing its performance).

If going for EFR32MG2x then do like IKEA and baying one certificated calibrated module from Silabs. I have not getting time dumping tuya modules but it can bein that they have doing the same mistake but i need verifying that then i have time.

Have you reading the last https://github.com/home-assistant/core/issues/48592 ?
Sniffing is working so the radio is working but dont knowing how well or if its one hardware or software problems.

**Comment by MattWestb** [comment]
### Silabs have start filtering IEEE 802.15.4 Ack packages in the radio encoder:

EZSP 6.7.9.0:
```
No.	Time	Protocol	Zigbee Src	Zigbee Dst	IEEE Src	IEEE Dst	Group Nr	ZBN Seq	ZBA Seq	ZDP Seq	Nwk Seq	Info
132	07:58:47,103837	IEEE 802.15.4	0x4b21	0x2646	0x4b21	0x2646					40	Data Request
133	07:58:47,105774	IEEE 802.15.4	N/A	N/A							40	Ack
134	07:58:48,125684	IEEE 802.15.4	0x4b21	0x2646	0x4b21	0x2646					41	Data Request
135	07:58:48,127495	IEEE 802.15.4	N/A	N/A							41	Ack
136	07:58:48,671593	ZigBee	0x2646	Broadcast	0x2646	0xffff		4			119	Link Status
137	07:58:51,644444	ZigBee HA	0x4b21	Broadcast	0x4b21	0x2646	0x0002	148	145		42	ZCL OnOff: Toggle, Seq: 54
138	07:58:51,646774	IEEE 802.15.4	N/A	N/A							42	Ack
139	07:58:51,696765	ZigBee HA	0x4b21	Broadcast	0x2646	0xffff	0x0002	148	145		120	ZCL OnOff: Toggle, Seq: 54
```

EZSP 6.9.2.0:
```
No.	Time	Protocol	Zigbee Src	Zigbee Dst	IEEE Src	IEEE Dst	Group Nr	ZBN Seq	ZBA Seq	ZDP Seq	Nwk Seq	Info
214	10:19:05,066709	ZigBee	0x5550	0x0000	0x5550	0x0000		128	27		251	APS: Ack, Dst Endpt: 1, Src Endpt: 1
215	10:19:05,481831	IEEE 802.15.4	0x4b21	0x2646	0x4b21	0x2646					114	Data Request
216	10:19:05,744141	ZigBee	0xefd5	Broadcast	0xefd5	0xffff		84			28	Link Status
217	10:19:06,507005	IEEE 802.15.4	0x4b21	0x2646	0x4b21	0x2646					115	Data Request
218	10:19:07,533380	IEEE 802.15.4	0x4b21	0x2646	0x4b21	0x2646					116	Data Request
219	10:19:07,666773	ZigBee	0x5550	Broadcast	0x5550	0xffff		130			252	Link Status
220	10:19:07,716872	ZigBee HA	0x4b21	Broadcast	0x4b21	0x2646	0x0002	169	155		117	ZCL OnOff: Toggle, Seq: 61
221	10:19:07,771770	ZigBee HA	0x4b21	Broadcast	0x2646	0xffff	0x0002	169	155		241	ZCL OnOff: Toggle, Seq: 61
222	10:19:07,781830	ZigBee HA	0x2646	0x0000	0x2646	0x0000		20	74		242	ZCL: Report Attributes, Seq: 32
```
In EZSP 6.7.9.0 you is  receiving all Ack from the IEEE 802.15.4 layer and ca see if some is missing.
In EZSP 6.9..2.0 is only passing the IEEE 802.15.4 Data Request and no Acks is coming from the radio decoder.

In Silabs Network analyzer you can see all IEEE 802.15.4 packages butt the acks is flagged as blocked so is not showing in the main view with exceptions if the program is thinking the Ack is addressed to it self (the same NWK as the PTI sniffing is being done) then is being showed with NWK address for sender and receiver (expected then the package dont have it only PAN-ID and sequence number).

![EZSP-ACK01](https://user-images.githubusercontent.com/49618193/115956021-c8a57d00-a4fa-11eb-9423-9d0131acc40d.PNG)
The PTI sniffing node is 2646 so its showing the ack as OK and the other is flagged as blocked (The sniff is made on one MG22 board not on normal ones but looks working OK).

The both wireshark is made on one tuya/LIDL LED stripe controller MG21 (same model as Sonoff ZBB) with self made NCP image (is for the hunt of Sonoff bug).

Some have saying that the MG22 is having on separate CPU for the radio but i need checking that and then i thinking MG21 is somthing in between with some strange side effects / bugs that  can being behind the "Sonoff" battery killings.

So if you like doing  on more detailed sniffing you shall have on EZSP 6.7 or older firmware in you sniffing device and if you like have one "censured" sniff going for on EZSP 6.9 (i have not looking if 8.X is filtering IEEE packages but you can trying if you like to knowing).

I shall testing if its doing the same with My Billy with EZSP 6.9.2.0 for seeing if its only EZSP version that is making the difference or if its hardware dependent.

**Reply by MattWestb** [reply]
Billy EZSP with 6.92.0 is doing the same in sniffing mode  as MG21/22 so its not hardware depended only software version.

So if like to see the IEEE 801.15.4 Acks you need using EZSP 6.7 (perhaps 8.x) or earlier EZSP or you dont see the acks.

**Comment by MattWestb** [comment]
Wiresharks latest update (June 2021) have starting checking the CRC in the packages and EZSP 8.x and newer is having one bug that is not adding the correct frame CRC to the raw frame so wireshark is complaining on all packets its receiving and not decoding them.

Silabs have issues one bug report (UID 710289) but then / if its being fixed is not is not known.

I think its the same thing with the missing 15.4 Acks that is not working  in 8.x and newer stacks.

So if using wireshark and EZSP device for sniffing its only EZSP 6.7.9.0 or older for getting it working with wireshark (both with bellows dump and com.zsmartsystems.zigbee.sniffer).

Closed bellows issue https://github.com/zigpy/bellows/issues/425 for info.

**Reply by MattWestb** [reply]
Its looks like `com.zsmartsystems.zigbee.sniffer` with six ten is working with wireshark on local comports but  i dont knowing way.

**Reply by puddly** [reply]
You can configure Wireshark to ignore the FCS and dissect every packet (uncheck `Dissect only good FCS`):

<img width="531" alt="image" src="https://user-images.githubusercontent.com/32534428/125815313-f13a0b5a-e901-49ed-b3c3-600b85acdc9e.png">

**Reply by MattWestb** [reply]
Sorry but its looks it not working ;-(((
![WS01](https://user-images.githubusercontent.com/49618193/125819128-d8c6cc19-77ca-44e1-907e-1e053a3ad207.PNG)
I trying updating WS then its have released one new version

But i like the option "Enable ACK tracking" then its possible see the source and destinations of all 25.4 packages also the ACKs :-)))))

**Comment by MattWestb** [comment]
With little [patches](https://github.com/zigpy/bellows/pull/428) added its now looks like this with six ten:
```
bellows -d socket://192.168.2.127:6638 info
[84:2e:14:ff:fe:0f:54:89]
[0x0000]
[<EmberNetworkStatus.JOINED_NETWORK: 2>]
[<EmberStatus.SUCCESS: 0>, <EmberNodeType.COORDINATOR: 1>, EmberNetworkParameters(extendedPanId=9c:df:71:25:83:34:20:0f, panId=0xc005, radioTxPower=8, radioChannel=16, joinMethod=<EmberJoinMethod.USE_MAC_ASSOCIATION: 0>, nwkManagerId=0x0000, nwkUpdateId=0, channels=<Channels.ALL_CHANNELS: 134215680>)]
[<EmberStatus.SUCCESS: 0>, EmberCurrentSecurityState(bitmask=<EmberCurrentSecurityBitmask.TRUST_CENTER_USES_HASHED_LINK_KEY|64|32|HAVE_TRUST_CENTER_LINK_KEY|GLOBAL_LINK_KEY: 244>, trustCenterLongAddress=84:2e:14:ff:fe:0f:54:89)]
Manufacturer: 
Board name: 
EmberZNet version: 6.10.0.0 build 169
```
Its the LIDL (tuya) LED strips RGBWW controller with one nice  MG21 module but no custom data is sett but CTune is made in the factory.

And dumping some frames:
```
bellows -d socket://192.168.2.127:6638 dump -c 15 -w dmp11.cap 

Capture started
^C
Captured 58 frames in 36.14s
```
and Wireshark is writing:
```
Frame 33: 52 bytes on wire (416 bits), 52 bytes captured (416 bits)
IEEE 802.15.4 Data, Dst: 0x0000, Src: 0x6701
ZigBee Network Layer Data, Dst: 0x0000, Src: 0x6701
ZigBee Application Support Layer Data, Dst Endpt: 1, Src Endpt: 1
    Frame Control Field: Data (0x00)
    Destination Endpoint: 1
    Cluster: Level Control (0x0008)
    Profile: Home Automation (0x0104)
    Source Endpoint: 1
    Counter: 105
ZigBee Cluster Library Frame, Command: Report Attributes, Seq: 18
    Frame Control Field: Profile-wide (0x08)
    Sequence Number: 18
    Command: Report Attributes (0x0a)
    Attribute Field
        Attribute: Current Level (0x0000)
        Data Type: 8-Bit Unsigned Integer (0x20)
        Current Level: 254
```
So some creative patching can doing great things :-)))

Great thanks @puddly !!!!

**Comment by pipiche38** [comment]
Do I need to have a special firmware to run Elelabs for sniffing ?

Other question, can we redirect to stdout instead of writing into a file ? does `-w -` works ?

[domoticz@rasp ~]$` bellows -d /dev/serial/by-id/usb-1a86_USB2.0-Serial-if00-port0 dump  -c 15 -w file.dump`

> Exception in callback ThreadsafeProxy.__getattr__.<locals>.func_wrapper.<locals>.check_result_wrapper() at /usr/local/lib/python3.9/site-packages/bellows/thread.py:97
> handle: <Handle ThreadsafeProxy.__getattr__.<locals>.func_wrapper.<locals>.check_result_wrapper() at /usr/local/lib/python3.9/site-packages/bellows/thread.py:97>
> Traceback (most recent call last):
>   File "/usr/lib/python3.9/asyncio/events.py", line 80, in _run
>     self._context.run(self._callback, *self._args)
>   File "/usr/local/lib/python3.9/site-packages/bellows/thread.py", line 98, in check_result_wrapper
>     result = call()
>   File "/usr/local/lib/python3.9/site-packages/bellows/ezsp/__init__.py", line 258, in frame_received
>     self._protocol(data)
> TypeError: 'NoneType' object is not callable
> 
> 
> Traceback (most recent call last):
>   File "/usr/lib/python3.9/asyncio/tasks.py", line 490, in wait_for
>     return fut.result()
> asyncio.exceptions.CancelledError
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/bellows", line 8, in <module>
>     sys.exit(main())
>   File "/usr/lib/python3.9/site-packages/click/core.py", line 829, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/lib/python3.9/site-packages/click/core.py", line 782, in main
>     rv = self.invoke(ctx)
>   File "/usr/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/usr/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/lib/python3.9/site-packages/click/core.py", line 610, in invoke
>     return callback(*args, **kwargs)
>   File "/usr/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "/usr/local/lib/python3.9/site-packages/bellows/cli/dump.py", line 30, in dump
>     loop.run_until_complete(_dump(ctx, channel, outfile))
>   File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
>     return future.result()
>   File "/usr/local/lib/python3.9/site-packages/bellows/cli/dump.py", line 60, in _dump
>     s = await util.setup(ctx.obj["device"], ctx.obj["baudrate"])
>   File "/usr/local/lib/python3.9/site-packages/bellows/cli/util.py", line 118, in setup
>     await s.reset()
>   File "/usr/local/lib/python3.9/site-packages/bellows/ezsp/__init__.py", line 98, in reset
>     await self._gw.reset()
>   File "/usr/local/lib/python3.9/site-packages/bellows/uart.py", line 223, in reset
>     return await asyncio.wait_for(self._reset_future, timeout=RESET_TIMEOUT)
>   File "/usr/lib/python3.9/asyncio/tasks.py", line 492, in wait_for
>     raise exceptions.TimeoutError() from exc
> asyncio.exceptions.TimeoutError

**Reply by Hedda** [reply]
> Do I need to have a special firmware to run Elelabs for sniffing ?

Maybe as a workaround try a slightly older firmware based on EmberZNet 6.7.x if you are currently using a newer firmware? 

https://github.com/zha-ng/EZSP-Firmware/tree/master/Elelabs-ELU013

https://github.com/grobasoz/zigbee-firmware/tree/master/EFR32%20Series%201

Regardless suggest that post about problem as a separate new issue to bellows for tracking with info on exact firmware version using:

https://github.com/zigpy/bellows/issues

You should at least not need special firmware according to this section about how to sniff using HUSBZB-1 with standard firmware:

https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html

https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-husbzb-1-stick

Could also try that "com.zsmartsystems.zigbee.sniffer" sniffer driver from zsmartsystems as a ZigBee sniffer interface for Wireshark:

https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer

**Reply by puddly** [reply]
This is what I use:

```bash
bellows -d /dev/... dump -c 15 -w /dev/stdout | wireshark -k -S -i -
```

Your adapter likely needs `--baudrate 115200` to work, as bellows defaults to something else.

**Comment by MattWestb** [comment]
`com.zsmartsystems.zigbee.sniffer` is piping the stream to witeshark and on the terminal but i think bellows -dump can only saving to file.
Bellows is supporting all modern EZSP firmware but newer firmware (6.9.x )is filtering out 802.15.4 acks so you cant see them if you like deep digging in the communication you need using EZSP 6.7.10.x or older.
I have not using bellows for sniffing for nearly one year (with tuya ZBGW) but i think its no changes made that have braking it but i cant grantee it working OK in current version.
